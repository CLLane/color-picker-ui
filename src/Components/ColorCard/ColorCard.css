.hex-code__input {
  background: rgb(0,0,0,0);
  border: rgb(0,0,0,.2) solid 1px;
  font-size: 20px;
  font-weight: bold;
  box-shadow: 1px 1px 2.5px black;
  height: 25px;
}

.hex-code__input::placeholder {
  color: black;
}