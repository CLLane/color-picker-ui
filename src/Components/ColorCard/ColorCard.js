import React from 'react';
import './ColorCard.css';
import unlockedIcon from '../../Images/unlockedIcon.svg';
import lockedIcon from '../../Images/lockedIcon.svg';

export const ColorCard = ({ card, toggleColorLock}) => {
  let cardStyle = {
    background: card.color,
    height: "30vh",
    width: "30vh",
    borderRadius: "8px",
    display: "flex",
    flexDirection: "column",
    justifyContent: "flex-end",
    marginRight: "5px",
    marginLeft: '5px',
    padding: "20px",
    boxShadow: "4px 4px 20px black"
  };

  const lockImage = card.locked ? lockedIcon : unlockedIcon;
  return (
    <article style={cardStyle}>
      <img
        src={lockImage}
        alt="unlocked"
        onClick={() => toggleColorLock(card.index)}
      />
      <input className='hex-code__input' disabled={true} type="text" placeholder={card.color}></input>
    </article>
  );
}

export default ColorCard;