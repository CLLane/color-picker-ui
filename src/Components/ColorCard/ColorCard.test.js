import React from 'react';
import { shallow } from 'enzyme';
import ColorCard from './ColorCard';
import './ColorCard.css';

describe('Color Card', () => {
  let wrapper;
  let mockCard;
  let toggleColorMock;

  beforeEach(() => {
    mockCard = {
      color: '#FFFFFF',
      locked: false, 
      index: 0
    };
    toggleColorMock = jest.fn();
    wrapper = shallow(<ColorCard 
      card={mockCard} 
      toggleColorLock={toggleColorMock} 
      key={0}
    />);
  });

  it('should match the snapshot when the card is not locked', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when the card is locked', () => {
    mockCard = { color: '#000000', locked: true };
    wrapper = shallow(<ColorCard 
      card={mockCard} 
      toggleColorLock={jest.fn} 
      key={0}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('call the toggle lock function when the button is clicked', () => {
    wrapper.find('img').simulate('click');
    expect(toggleColorMock).toHaveBeenCalledWith(0);
  });
});


