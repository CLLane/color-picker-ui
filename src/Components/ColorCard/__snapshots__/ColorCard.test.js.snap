// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Card should match the snapshot when the card is locked 1`] = `
<article
  style={
    Object {
      "background": "#000000",
      "borderRadius": "8px",
      "boxShadow": "4px 4px 20px black",
      "display": "flex",
      "flexDirection": "column",
      "height": "30vh",
      "justifyContent": "flex-end",
      "marginLeft": "5px",
      "marginRight": "5px",
      "padding": "20px",
      "width": "30vh",
    }
  }
>
  <img
    alt="unlocked"
    onClick={[Function]}
    src="lockedIcon.svg"
  />
  <input
    className="hex-code__input"
    disabled={true}
    placeholder="#000000"
    type="text"
  />
</article>
`;

exports[`Color Card should match the snapshot when the card is not locked 1`] = `
<article
  style={
    Object {
      "background": "#FFFFFF",
      "borderRadius": "8px",
      "boxShadow": "4px 4px 20px black",
      "display": "flex",
      "flexDirection": "column",
      "height": "30vh",
      "justifyContent": "flex-end",
      "marginLeft": "5px",
      "marginRight": "5px",
      "padding": "20px",
      "width": "30vh",
    }
  }
>
  <img
    alt="unlocked"
    onClick={[Function]}
    src="unlockedIcon.svg"
  />
  <input
    className="hex-code__input"
    disabled={true}
    placeholder="#FFFFFF"
    type="text"
  />
</article>
`;
